frontend/server/src/Controllers/Session.php

```diff
@@ -327,9 +329,38 @@ private static function getCurrentSessionImpl(
             if (is_null($currentUser)) {
                 throw new \OmegaUp\Exceptions\NotFoundException('userNotExist');
             }
+            $now = \OmegaUp\Time::get();
+            if (
+                !is_null(
+                    $currentUser->birth_date
+                ) && strtotime(
+                    $currentUser->birth_date
+                ) >= strtotime(
+                    '-13 year',
+                    $now
+                ) && !$currentUser->parent_verified
+            ) {
+                if (is_null($currentUser->parent_email_verification_deadline)) {
+                    throw new \OmegaUp\Exceptions\InvalidParameterException(
+                        'parameterNotFound',
+                        'parent_email_verification_deadline'
+                    );
+                }
+                $diff = $currentUser->parent_email_verification_deadline->time - $now;
+                $pendingDaysForVerification = intval(
+                    floor(
+                        $diff / (60 * 60 * 24)
+                    )
+                );
+                if ($pendingDaysForVerification < 0) {
+                    throw new \OmegaUp\Exceptions\NotFoundException(
+                        'U13HasNotBeenVerifiedForTheirParent'
+                    );
+                }
+            }
             $email = !is_null($currentUser->main_email_id) ?
                 \OmegaUp\DAO\Emails::getByPK($currentUser->main_email_id) :
-                null;
+                 null;
```

Comment:
```suggestion
                null;
```

try to keep the unrelated whitespace changes to a minimum

Reply:
Sure!